crosshair/libimpl/collectionslib_test.py

```diff
@@ -12,28 +13,102 @@
 from crosshair.util import set_debug
 
 class CollectionsLibTests(unittest.TestCase):
-    # Remove these TODO prefixes to enable the tests:
-    def TODO_test_deque_len_ok(self) -> None:
-        def f(l: Deque[int]) -> Deque[int]:
-            '''
-            post: len(_) == len(__old__.l) + 1
-            '''
-            l.append(42)
-            return l
-        # Right now, this returns unknown rather than ok, because creation of
-        # the deque forces the list length to become concrete.
-        self.assertEqual(*check_ok(f))
-
-    def TODO_test_deque_len_fail(self) -> None:
-        def f(l: Deque[int]) -> Deque[int]:
-            '''
-            pre: len(l) > 0
-            post: len(l) != 222
-            '''
-            return l
-        # Right now, this returns unknown rather than fail, because creation of
-        # the deque forces the list length to become concrete.
-        self.assertEqual(*check_fail(f))
+
+    def setUp(self):
+        self.test_list = ListBasedDeque([1, 2, 3, 4, 5])
```

Comment:
Personal preference thing (so you should feel free to ignore if you like): I tend to bias towards repeating the test data in each test. It adds an extra line to a lot of tests, but makes them easier to read.